pkg/endpoint/bpf.go

```diff
@@ -1124,6 +1125,12 @@ func (e *Endpoint) syncPolicyMap() error {
 	if e.realizedPolicy != e.desiredPolicy {
 		errors := 0
 
+		// Add policy map entries before deleting to avoid transient drops
+		err := e.addPolicyMapDelta()
+		if err != nil {
+			errors++
+		}
+
 		// Delete policy keys present in the realized state, but not present in the desired state
 		for keyToDelete := range e.realizedPolicy.PolicyMapState {
 			// If key that is in realized state is not in desired state, just remove it.
```

Comment:
Can this loop ever delete policy that was added on line 1129?

Reply:
No, as `addPolicyMapDelta` adds keys present in `desiredPolicy` and in this loop we explicitly only delete keys that are NOT present in `desiredPolicy`.